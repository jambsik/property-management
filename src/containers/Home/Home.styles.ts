import { emerald } from './../../theme/colors';

export const homeStylesRoot = {
    width: '100%',
    height: '100%',
    backgroundColor: emerald,
    display: 'flex',
    flexDirection: 'column' as const,
    justifyContent: 'center',
    alignItems: 'center',
};

export const homeGetStartedStyles = {
    marginTop: '24px',
};
