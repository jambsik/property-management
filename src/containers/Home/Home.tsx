import React from 'react';
import { Box, Button, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import { homeGetStartedStyles, homeStylesRoot } from './Home.styles';
import { APP_ROUTES_PATH } from '../../config/routes';

const Home = () => {
    const { t } = useTranslation();

    return (
        <Box sx={homeStylesRoot}>
            <Typography variant="h1" color="primary">
                {t('home.title')}
            </Typography>
            <Link to={APP_ROUTES_PATH.mockView}>
                <Button sx={homeGetStartedStyles} variant="contained" size="medium" color="secondary">
                    {t('home.get_started')}
                </Button>
            </Link>
        </Box>
    );
};

export default Home;
