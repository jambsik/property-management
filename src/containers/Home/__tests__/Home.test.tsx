import React from 'react';
import { render } from '@testing-library/react';
import Home from '../Home';
import { BrowserRouter } from 'react-router-dom';
describe('Home container test', () => {
    it('Should render component without errors', () => {
        const { container } = render(
            <BrowserRouter>
                <Home />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
