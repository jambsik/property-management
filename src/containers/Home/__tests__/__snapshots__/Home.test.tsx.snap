// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container test Should render component without errors 1`] = `
<div>
  <div
    class="MuiBox-root css-srv7rf"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-utprum-MuiTypography-root"
    >
      home.title
    </h1>
    <a
      href="/mock-view"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1nhhl2i-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        home.get_started
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</div>
`;
